File: dotnet/Allmark/Render/ThematicBreakRenderer.cs
namespace Allmark.Render;

using Allmark.Types;

public static class ThematicBreakRenderer
{
    public static OutputRenderer Create()
    {
        return new OutputRenderer
        {
            Name = "thematic_break",
            Render = Render,
        };
    }

    public static void Render(MarkdownNode node, RendererState state, bool? first = null, bool? last = null, bool? decode = true)
    {
        RenderUtils.StartNewLine(node, state);
        state.Output.Append("<hr />");
        RenderUtils.EndNewLine(node, state);
    }
}

File: zig/src/render/thematicBreakRenderer.zig
const std = @import("std");
const MarkdownNode = @import("../types/MarkdownNode.zig").MarkdownNode;
const RendererState = @import("../types/RendererState.zig").RendererState;
const Renderer = @import("../types/Renderer.zig").Renderer;
const renderSelfClosedTag = @import("renderSelfClosedTag.zig").renderSelfClosedTag;

pub fn render(node: *const MarkdownNode, state: *RendererState, decode: ?bool) void {
    _ = decode;
    renderSelfClosedTag(node, state, "hr");
}

pub const thematicBreakRenderer = Renderer{
    .name = "thematic_break",
    .render = render,
};
